packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,710 @@
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers S3 and Log Policies', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              's3:PutObject',
+              's3:GetObject',
+              's3:ListBucket',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['arn:aws:logs:*:*:*'],
+            actions: [
+              'logs:PutLogEvents',
+              'logs:CreateLogStream',
+              'logs:DescribeLogGroups',
+              'logs:DescribeLogStreams',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Job Run Policy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:StartJobRun',
+              'emr-containers:ListJobRuns',
+              'emr-containers:DescribeJobRun',
+              'emr-containers:CancelJobRun',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Monitoring Policy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:DescribeJobRun',
+              'elasticmapreduce:CreatePersistentAppUI',
+              'elasticmapreduce:DescribePersistentAppUI',
+              'elasticmapreduce:GetPersistentAppUIPresignedURL',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              's3:GetObject',
+              's3:ListBucket',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'logs:Get*',
+              'logs:DescribeLogGroups',
+              'logs:DescribeLogStreams',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    this.updateRoleTrustPolicy(jobExecutionRole.roleName);
+
+    return jobExecutionRole;
+  }
+
+  /**
+   * Creates custom resources that call describeJobRun and executes an lambda
+   * that calls update-role-trust-policy using the inputs.
+   *
+   * @param roleName - provided from automatically generated name
+   */
+  private updateRoleTrustPolicy(roleName: string) {
+
+    // first create a custom resource to retrieve the eks namespace and eks cluster output from describe virtual cluster
+    const descVirtClust = new cr.AwsCustomResource(this, 'EMR Containers DescribeVirtualCluster SDK caller', {
+      onCreate: {
+        service: 'EMRcontainers',
+        action: 'describeVirtualCluster',
+        parameters: {
+          id: this.props.virtualClusterId.value,
```

Comment:
need to make sure none of this is happening if this is a JSON path *and* that we make that clear in documentation

Reply:
By this you mean ensuring prior to the automatic creation of the job role the virtualClusterId is not a json path?

Reply:
I'm thinking of adding validation for this because customers could potentially have an encoded json path for the VC id and also not provide the execution role.

Reply:
Something along the lines of this

```ts 
if (props.executionRole === undefined
      && sfn.JsonPath.isEncodedJsonPath(props.virtualClusterId.value)) {
      throw new Error('Execution role cannot be undefined when the virtual cluster id is an encoded JSON path');
    }
```